server/controllers/userController.js

```diff
@@ -106,4 +109,61 @@ export default class UsersController {
       return next(err);
     }
   }
+
+  /**
+   *
+   * Edit user profile
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} Object containing user profile
+   * @memberof UsersController
+   */
+  static async editUserProfile(req, res, next) {
+    try {
+      const data = req.body;
+      const { id } = req.params;
+      const { userId } = req;
+      if (parseInt(id, 10) !== userId) {
+        return badPostRequest(res, 401, {
+          message: 'Unauthorized access'
+        });
+      }
+      const userProfile = await editProfile(id, data);
+      if (!userProfile) {
+        return badPostRequest(res, 400, {
+          message: 'Profile failed to update'
+        });
+      }
+      return successfulRequest(res, 200, { ...userProfile });
+    } catch (err) {
+      return next(err);
+    }
+  }
+
+  /**
+   *
+   * View user profile
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} Object containing user profile
+   * @memberof UsersController
+   */
+  static async viewUserProfile(req, res, next) {
+    try {
+      const { id } = req.params;
+      const userProfile = await viewProfile(id);
+      if (!userProfile) {
+        return badPostRequest(res, 404, {
+          message: 'Profile not found'
+        });
+      }
+      return successfulRequest(res, 200, { ...userProfile });
```

Comment:
Great work here Nero! I like how you added the swagger definition for the UserProfile, however, some of your responses, don't exactly follow the specification as it's missing a `data` field when sending back a response with data. Please refer to [this](https://github.com/omniti-labs/jsend) specification for more guidance. Once again, Nice job!

Reply:
Forgive me, I just noticed the screenshot had the right responses! Great Job Nero